test/Sema/diag_unintended_optional_behavior_swift_5.swift

```diff
@@ -28,64 +25,23 @@ class D {
-  _ = takeAny(C()[0], C()[1.0]) // expected-warning {{expression implicitly coerced from 'Int?' to 'Any'}}
-  // expected-note@-1 {{provide a default value to avoid this warning}}{{21-21= ?? <#default value#>}}
-  // expected-note@-2 {{force-unwrap the value to avoid this warning}}{{21-21=!}}
-  // expected-note@-3 {{explicitly cast to 'Any' with 'as Any' to silence this warning}}{{21-21= as Any}}
-  // expected-warning@-4 {{expression implicitly coerced from 'Any?' to 'Any'}}
-  // expected-note@-5 {{provide a default value to avoid this warning}}{{31-31= ?? <#default value#>}}
-  // expected-note@-6 {{force-unwrap the value to avoid this warning}}{{31-31=!}}
-  // expected-note@-7 {{explicitly cast to 'Any' with 'as Any' to silence this warning}}{{31-31= as Any}}
-  _ = takeAny(D(), D()) // expected-warning {{expression implicitly coerced from 'D?' to 'Any'}}
-  // expected-note@-1 {{provide a default value to avoid this warning}}{{18-18= ?? <#default value#>}}
-  // expected-note@-2 {{force-unwrap the value to avoid this warning}}{{18-18=!}}
-  // expected-note@-3 {{explicitly cast to 'Any' with 'as Any' to silence this warning}}{{18-18= as Any}}
-  // expected-warning@-4 {{expression implicitly coerced from 'D?' to 'Any'}}
-  // expected-note@-5 {{provide a default value to avoid this warning}}{{23-23= ?? <#default value#>}}
-  // expected-note@-6 {{force-unwrap the value to avoid this warning}}{{23-23=!}}
-  // expected-note@-7 {{explicitly cast to 'Any' with 'as Any' to silence this warning}}{{23-23= as Any}}
+  _ = takeAny(a, b)
+  _ = takeAny(returningIUO(), C().returningIUO())
+  _ = takeAny(C().a, C().b)
+  _ = takeAny(C()[0], C()[1.0])
+  _ = takeAny(D(), D())
 
   _ = takeAny(a as Any, b as Any)
 }
 
 func warnIUOToOptionalAnyCoercion(_ a: Int!, _ b: Any?!, _ c: Int??!, _ d: Any???!) {
-  takesOptionalAny(a, b) // expected-warning {{expression implicitly coerced from 'Any??' to 'Any?'}}
-  // expected-note@-1 {{provide a default value to avoid this warning}}{{24-24= ?? <#default value#>}}
-  // expected-note@-2 {{force-unwrap the value to avoid this warning}}{{24-24=!}}
-  // expected-note@-3 {{explicitly cast to 'Any?' with 'as Any?' to silence this warning}}{{24-24= as Any?}}
+  takesOptionalAny(a, b)
 
   takesOptionalAny(a, b ?? "")
   takesOptionalAny(a, b!)
   takesOptionalAny(a, b as Any?)
 
-  takesOptionalAny(c, d) // expected-warning {{expression implicitly coerced from 'Int???' to 'Any?'}}
-  // expected-note@-1 {{force-unwrap the value to avoid this warning}}{{21-21=!!}}
-  // expected-note@-2 {{explicitly cast to 'Any?' with 'as Any?' to silence this warning}}{{21-21= as Any?}}
-  // expected-warning@-3 {{expression implicitly coerced from 'Any????' to 'Any?'}}
-  // expected-note@-4 {{force-unwrap the value to avoid this warning}}{{24-24=!!!}}
-  // expected-note@-5 {{explicitly cast to 'Any?' with 'as Any?' to silence this warning}}{{24-24= as Any?}}
+  takesOptionalAny(c, d)
```

Comment:
Sorry for the delay, I think we should still diagnose this case because `Int??!` is effectively `Int??` which gets converted to `Any`.

Reply:
Do you mean we only allow conversions from, example, `T!` but not `T?!`, etc? In other words, if we have an ? before the !, then diagnose?

Reply:
Yes, that’s right.

Reply:
Thanks @xedin, I have made the changes :)